Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -1715,6 +1716,69 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_spliced)
     return return_array;
 }
 
+// 1.2.2.1.6 %TypedArray%.prototype.with ( index, value ), https://tc39.es/proposal-change-array-by-copy/#sec-%typedarray%.prototype.with
+JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::with)
+{
+    auto index = vm.argument(0);
+
+    // 1. Let O be the this value.
+    // 2. Perform ? ValidateTypedArray(O).
+    auto* typed_array = TRY(validate_typed_array_from_this(global_object));
+
+    // 3. Let len be O.[[ArrayLength]].
+    auto length = typed_array->array_length();
+
+    // 4. Let relativeIndex be ? ToIntegerOrInfinity(index).
+    auto relative_index = TRY(index.to_integer_or_infinity(global_object));
+
+    double actual_index = 0;
+    // 5. If relativeIndex ≥ 0, let actualIndex be relativeIndex.
+    if (relative_index >= 0)
+        actual_index = relative_index;
+    else
+        actual_index = length + relative_index;
+
+    Value value;
+    // 7. If O.[[ContentType]] is BigInt, set value to ? ToBigInt(value).
+    if (typed_array->content_type() == TypedArrayBase::ContentType::BigInt)
+        value = TRY(vm.argument(1).to_bigint(global_object));
+    // 8. Else, set value to ? ToNumber(value).
+    else
+        value = TRY(vm.argument(1).to_number(global_object));
+
+    // 9. If ! IsValidIntegerIndex(O, 𝔽(actualIndex)) is false, throw a RangeError exception.
+    if (!is_valid_integer_index(*typed_array, CanonicalIndex(CanonicalIndex::Type::Index, actual_index)))
+        return vm.throw_completion<RangeError>(global_object, ErrorType::TypedArrayInvalidTargetOffset, "finite");
```

Comment:
error message makes no sense, this isn't an offset, nor is it necessarily infinite if this check fails